record/record.go

```diff
@@ -391,6 +459,22 @@ func (w *Writer) Next() (io.Writer, error) {
 	return singleWriter{w, w.seq}, nil
 }
 
+// LastRecordOffset returns the offset in the underlying io.Writer of the last
+// record so far - the one created by the most recent Next call. It is the
+// offset of the first chunk header, suitable to pass to Reader.SeekRecord.
+//
+// If that io.Writer also implements io.Seeker, the return value is an absolute
+// offset, in the sense of io.SeekStart, regardless of whether the io.Writer
+// was initially at the zero position when passed to NewWriter. Otherwise, the
+// return value is a relative offset, being the number of bytes written between
+// the NewWriter call and any records written prior to the last record.
+func (w *Writer) LastRecordOffset() (int64, error) {
```

Comment:
If no records were written, LastRecordOffset() will return the zero value for w.lastRecordOffset.  I could initialize it to -1, and check for that, returning an error in that case. Or simply have LastRecordOffset return 0. Or document it as undefined. What do you think Nigel?